Open Judge System/Services/OJS.Services/Data/ParticipantScores/ParticipantScoresDataService.cs

```diff
@@ -167,6 +167,23 @@ public void Delete(IEnumerable<ParticipantScore> participantScoresEnumerable)
             this.participantsData.Update(participant);
         }
 
+        public void UpdateBySubmissionAndPoints(ParticipantScore participantScore, int? submissionId, int submissionPoints,
+            Participant participant, bool withSaveChanges)
+        {
+            //The submission TotalScoreSnapshotModifiedOn must be changed only if it is new submission in other way the results will not be ordered correctly.
+            bool shouldUpdateTotalScoreDate = submissionId != null && submissionId != participantScore.SubmissionId;
+            UpdateTotalScoreSnapshot(
+                participant, 
+                participantScore.Points, 
+                submissionPoints,
+                shouldUpdateTotalScoreDate);
+
+            participantScore.SubmissionId = submissionId;
+            participantScore.Points = submissionPoints;
```

Comment:
This is repetition of knowledge (from the overload above), which we can easily avoid.

Reply:
Fixed